packages/@aws-cdk/core/lib/cfn-parse.ts

```diff
@@ -519,9 +529,11 @@ export class CfnParser {
     if (isRef) {
       const refElement = this.options.finder.findRefTarget(refTarget);
       if (!refElement) {
-        throw new Error(`Element referenced in Fn::Sub expression with logical ID: '${refTarget}' was not found in the template`);
+        if  (!(refTarget in (this.parameters))) {
+          throw new Error(`Element referenced in Fn::Sub expression with logical ID: '${refTarget}' was not found in the template`);
+        }
       }
-      return leftHalf + CfnReference.for(refElement, 'Ref', true).toString() + this.parseFnSubString(rightHalf, map);
+      return leftHalf + this.specialCaseSubRefs(refTarget, false, refElement) + this.parseFnSubString(rightHalf, map);
```

Comment:
Sorry, why this change? `specialCaseSubRefs` is already called above, I don't understand why does it need to be called again?